zkvm/src/verifier.rs

```diff
@@ -10,33 +10,40 @@ use crate::point_ops::PointOp;
 use crate::signature::VerificationKey;
 use crate::types::*;
 
-use crate::vm::{Delegate, RunTrait, Tx, VerifiedTx, VM};
+use crate::vm::{Delegate, Tx, VerifiedTx, VM};
 
 pub struct Verifier<'a, 'b> {
     signtx_keys: Vec<VerificationKey>,
     deferred_operations: Vec<PointOp>,
     cs: r1cs::Verifier<'a, 'b>,
 }
 
-pub struct RunVerifier {
+pub struct VerifierRun {
     program: Vec<u8>,
     offset: usize,
 }
 
 impl<'a, 'b> Delegate<r1cs::Verifier<'a, 'b>> for Verifier<'a, 'b> {
-    type RunType = RunVerifier;
-
-    fn commit_variable(&mut self, com: &Commitment) -> (CompressedRistretto, r1cs::Variable) {
-        let point = com.to_point();
-        let var = self.cs.commit(point);
-        (point, var)
+    type RunType = VerifierRun;
+
+    fn commit_variable(
+        &mut self,
+        com: &Commitment,
+    ) -> Result<(CompressedRistretto, r1cs::Variable), VMError> {
+        let point = match com {
```

Comment:
don't we have `to_point` method for commitment?

Reply:
we do! but to mirror the prover, I wanted to fail if the commitment is `Open`